packages/preferences/README.md

```diff
@@ -0,0 +1,172 @@
+# Preferences
+
+Utilities for storing WordPress preferences.
+
+## Installation
+
+Install the module
+
+```bash
+npm install @wordpress/preferences --save
+```
+
+_This package assumes that your code will run in an **ES2015+** environment. If you're using an environment that has limited or no support for such language features and APIs, you should include [the polyfill shipped in `@wordpress/babel-preset-default`](https://github.com/WordPress/gutenberg/tree/HEAD/packages/babel-preset-default#polyfill) in your code._
+
+## Examples
+
+### Data store
+
+Preferences are persisted values of any kind.
+
+Set the default preferences for any features on initialization by dispatching an action:
+
+```js
+import { dispatch } from '@wordpress/data';
+import { store as preferencesStore } from '@wordpress/preferences';
+
+function initialize() {
+	// ...
+
+	dispatch( preferencesStore ).setDefaults(
+		'namespace/editor-or-plugin-name',
+		{
+			myBooleanFeature: true,
+		}
+	);
+
+	// ...
+}
+```
+
+Or the `get` selector to get a preference value, and the `set` action to update a preference to any value:
+
+```js
+wp.data
+	.select( 'core/preferences' )
+	.get( 'namespace/editor-or-plugin-name', 'myPreferenceName' ); // 1
+wp.data
+	.dispatch( 'core/preferences' )
+	.set( 'namespace/editor-or-plugin-name', 'myPreferenceName', 2 );
+wp.data
+	.select( 'core/preferences' )
+	.get( 'namespace/editor-or-plugin-name', 'myPreferenceName' ); // 2
+```
+
+Use the `toggle` action to flip a boolean preference between `true` and `false`:
+
+```js
+wp.data
+	.select( 'core/preferences' )
+	.get( 'namespace/editor-or-plugin-name', 'myPreferenceName' ); // true
+wp.data
+	.dispatch( 'core/preferences' )
+	.toggle( 'namespace/editor-or-plugin-name', 'myPreferenceName' );
+wp.data
+	.select( 'core/preferences' )
+	.get( 'namespace/editor-or-plugin-name', 'myPreferenceName' ); // false
+```
+
+### Components
+
+The `MoreMenuDropdown` and `MoreMenuPreferenceToggle` components help to implement an editor menu for changing preferences and feature values.
```

Comment:
While I think `MoreMenuPreferenceToggle` makes sense in this package (probably better renamed as just `PreferenceToggle`), I have some trouble understanding the need of `MoreMenuDropdown` in the context of the preferences package.

Reply:
This sounds like sharing code just to share code and avoid duplication. In general, I feel like this might not always be the right argument for code sharing as it makes us create wrong abstractions. I'm fine with a sharable DropdownMenu but this one particularly sounds more suited for the "interface" package instead. It has nothing to do with "preferences".

Reply:
👍  Interface sounds like a good option. After all there's a bunch of options in this menu that's not related to preferences. 

Some of the plugin extensibility that's loosely connected with the dropdown lives in interface, so that works.

Reply:
That's all done now. I've renamed `MoreMenuPreferenceToggle` to `PreferenceToggleMenuItem`. It's a bit longer than the name suggested, but in future we may add another toggle component for the preferences modal.